.github/workflows/build-and-test.yml

```diff
@@ -943,6 +942,104 @@ jobs:
         if: ${{ failure() && runner.debug }}
         uses: canonical/action-tmate@main
 
+  juju-adopt-ceph-test:
```

Comment:
Fwiw we have pytest based functests now in tests/functests, nothing to block over but we could consider using these instead of shell